disco_aws_automation/disco_deploy.py

```diff
@@ -559,6 +547,66 @@ def handle_blue_green_ami(self, pipeline_dict, ami, old_group,
             self._disco_elb.delete_elb(hostclass, testing=True)
         return False
 
+    def _create_scaling_schedule(self, pipeline_dict, group_name=None, hostclass=None):
+        """ Create scaling schedules from the pipeline dictionary """
+        desired_size = pipeline_dict.get("desired_size", 1)
+        min_size = pipeline_dict.get("min_size", size_as_minimum_int_or_none(desired_size))
+        max_size = pipeline_dict.get("max_size", 0) or size_as_maximum_int_or_none(desired_size)
+
+        self._disco_aws.create_scaling_schedule(
+            min_size,
+            desired_size,
+            max_size,
+            group_name=group_name,
+            hostclass=hostclass
+        )
+
+    def _generate_final_pipeline(self, pipeline_dict, old_group, ami):
+        """Generate pipeline with sizing for post deployment"""
+        new_config = copy.deepcopy(pipeline_dict)
+        new_config["sequence"] = 1
+        new_config["smoke_test"] = "no"
+        new_config["ami"] = ami.id
+
+        # If there is an already existing ASG, use its sizing. Otherwise, use the pipeline's sizing or a
+        # reasonable default.
+        if old_group:
+            desired_size = old_group.desired_capacity
+            max_size = old_group.max_size
+            min_size = old_group.min_size
+        else:
+            # The 'or 1' is because some people set their desired size to 0 in their pipeline.
+            desired_size = int(size_as_maximum_int_or_none(
+                pipeline_dict.get("desired_size", 1) or 1
+            ))
+            min_size = int(size_as_minimum_int_or_none(
+                pipeline_dict.get("min_size", 0)
+            ))
+            # The 'or' on max_size is here for the same reason. So if it's 0, just set it to desired_size so
+            # its a valid entry...
+            max_size = int(size_as_maximum_int_or_none(
+                pipeline_dict.get("max_size", desired_size) or desired_size
+            ))
+
+        new_config["desired_size"] = desired_size
+        new_config["min_size"] = min_size
+        new_config["max_size"] = max_size
+
+        return new_config
+
+    def _generate_deploy_pipeline(self, deployment_strategy, pipeline_dict, old_group, ami):
+        """Generate pipeline with sizing for deployment"""
+        new_config = self._generate_final_pipeline(pipeline_dict, old_group, ami)
```

Comment:
For blue/green, they're really the same. It's only for classic deployment that they have to differ.